### PR TITLE
Add coliship product code in shipping methods

### PR DESCRIPTION
**Pull request by maximehuran**
Be able to set a Coliship Product Code on Shipping Method form : 

image

This one will be use if it's set during the export.
If not, the old default value `COLD` still be returned.

### CODE REVIEW

File: tests/Application/src/Entity/Shipping/ShippingMethod.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use MonsieurBiz\SyliusColishipPlugin\Entity\Shipping\ColishipShippingMethodTrait;
+use Sylius\Component\Core\Model\ShippingMethod as BaseShippingMethod;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="sylius_shipping_method")
+ */
+class ShippingMethod extends BaseShippingMethod implements ColishipShippingMethodInterface
+{
+    use ColishipShippingMethodTrait;
```
Comment (jacquesbh) [inline tests/Application/src/Entity/Shipping/ShippingMethod.php:27]: Maybe we can add a notice in the readme related to this change?

File: src/Form/Extension/Shipping/ShippingMethodTypeExtension.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class ShippingMethodTypeExtension extends AbstractTypeExtension
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('colishipProductCode', TextType::class, [
+                'label' => 'monsieurbiz_coliship.form.shipping_method.coliship_product_code',
+                'help' => 'monsieurbiz_coliship.form.shipping_method.coliship_product_code_help',
+                'required' => false,
+            ])
```
Comment (jacquesbh) [inline src/Form/Extension/Shipping/ShippingMethodTypeExtension.php:30]: No validation at all?

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: tests/Application/src/Entity/Shipping/ShippingMethod.php (hunk 1 above)
Comment (maximehuran) [reply tests/Application/src/Entity/Shipping/ShippingMethod.php:27]: The README is not defined yet

File: src/Form/Extension/Shipping/ShippingMethodTypeExtension.php (hunk 2 above)
Comment (maximehuran) [reply src/Form/Extension/Shipping/ShippingMethodTypeExtension.php:30]: What do you propose ?
The field is not required and there is no special validation to apply on it.

**Comment by jacquesbh** [comment]
Thank you @maximehuran !

**Pull request merged**